src/Services/AllergyIntoleranceService.php

```diff
@@ -0,0 +1,180 @@
+            $row['practitioner'] = $row['practitioner'] ?
+                UuidRegistry::uuidToString($row['practitioner']) :
+                $row['practitioner'];
+            if ($row['diagnosis'] != "") {
+                $this->addDiagnosis($row['diagnosis']);
+            }
+            $processingResult->addData($row);
+        }
+        return $processingResult;
+    }
+
+    /**
+     * Returns a single allergyIntolerance record by uuid.
+     * @param $uuid - The allergyIntolerance uuid identifier in string format.
+     * @return ProcessingResult which contains validation messages, internal error messages, and the data
+     * payload.
+     */
+    public function getOne($uuid)
+    {
+        $processingResult = new ProcessingResult();
+
+        $isValid = BaseValidator::validateId("uuid", "lists", $uuid, true);
+
+        if ($isValid !== true) {
+            $validationMessages = [
+                'uuid' => ["invalid or nonexisting value" => " value " . $uuid]
+            ];
+            $processingResult->setValidationMessages($validationMessages);
+            return $processingResult;
+        }
+
+        $sql = "SELECT lists.*,
+                        us.uuid as practitioner,
+                        patient.uuid as puuid,
+                        reaction.title as reaction_title,
+                        verification.title as verification_title
+                        FROM lists
+                        LEFT JOIN list_options as reaction ON reaction.option_id = lists.reaction
+                        LEFT JOIN list_options as verification ON verification.option_id = lists.verification
+                        LEFT JOIN users as us ON us.id = lists.referredby
+                        RIGHT JOIN patient_data as patient ON patient.pid = lists.pid
+                        WHERE type = 'allergy' AND lists.uuid = ?";
+
+        $uuidBinary = UuidRegistry::uuidToBytes($uuid);
+        $sqlResult = sqlQuery($sql, [$uuidBinary]);
+        $sqlResult['uuid'] = UuidRegistry::uuidToString($sqlResult['uuid']);
+        $sqlResult['puuid'] = UuidRegistry::uuidToString($sqlResult['puuid']);
+        $sqlResult['practitioner'] = $sqlResult['practitioner'] ?
+            UuidRegistry::uuidToString($sqlResult['practitioner']) :
+            $sqlResult['practitioner'];
+        if ($sqlResult['diagnosis'] != "") {
+            $this->addDiagnosis($sqlResult['diagnosis']);
+        }
+        $processingResult->addData($sqlResult);
+        return $processingResult;
+    }
+
+    private function addDiagnosis(&$diagnosis)
+    {
+        $diags = explode(";", $diagnosis);
+        $diagnosis = array();
+        foreach ($diags as $diag) {
+            $code = explode(':', $diag)[1];
+            $codeSql = "SELECT long_desc FROM icd10_dx_order_code WHERE active = 1
+                                AND valid_for_coding = '1'
+                                AND formatted_dx_code = '$code'";
```

Comment:
need to use binding to prevent sql-injection
```
AND formatted_dx_code = ?";
$codedesc = sqlQuery($codeSql, [$code]);
```

Reply:
actually, better to run these through the following function to collect the code descriptions:
https://github.com/openemr/openemr/blob/master/custom/code_types.inc.php#L680